model/modeldecoder/modeldecoderutil/labels.go

```diff
@@ -53,3 +74,56 @@ func NormalizeLabelValues(labels common.MapStr) common.MapStr {
 	}
 	return labels
 }
+
+func filterLabels(labels common.MapStr, fn func(interface{}) interface{}) common.MapStr {
+	result := common.MapStr{}
+	for k, v := range labels {
+		if typeValue := fn(v); typeValue != nil {
+			result[k] = typeValue
+		}
+	}
+	if len(result) > 0 {
+		return result
+	}
+	return nil
+}
+
+func filterStringLabels(v interface{}) interface{} {
+	switch v := v.(type) {
+	case string:
+		return v
+	case bool:
+		return strconv.FormatBool(v)
+	case []interface{}:
+		res := make([]interface{}, 0, len(v))
+		for i := range v {
+			if val := filterStringLabels(v[i]); val != nil {
+				res = append(res, val)
+			}
+		}
+		return res
+	}
+	return nil
+}
+
+func filterNumberLabels(v interface{}) interface{} {
+	switch v := v.(type) {
+	case float64:
+		return v
+	case int64:
+		return float64(v)
+	case int32:
+		return float64(v)
+	case int:
+		return float64(v)
```

Comment:
```suggestion
```

IIANM the JSON decoder always decodes numbers as json.Number, and we convert those to float64.